cg/apps/mip/parse_sampleinfo.py

```diff
@@ -1,71 +1,82 @@
+    dry_run_all: bool
+    log_file: str
+    outdata_dir: str
+    slurm_quality_of_service: Literal["low", "normal", "high"]
+    sample_info_file: str
+
+
+class MipBaseSampleinfo(BaseModel):
+    """This model is used when validating the mip samplinfo file"""
+
+    analysisrunstatus: str
+    analysis_date: str
+    case_id: str
+    family_id: str = None
+    human_genome_build: dict
+    mip_version: str
+    program: dict = None
+    recipe: dict = None
+
+
+def get_sample_data_from_config(config: MipBaseConfig) -> List[dict]:
     """Get sample data from config"""
     samples = []
-    for sample_id, analysis_type in config["analysis_type"].items():
+    for sample_id, analysis_type in config.analysis_type.items():
         samples.append({"id": sample_id, "type": analysis_type})
     return samples
 
 
 def parse_config(data: dict) -> dict:
-    """Parse MIP config file
+    """Validate and parse MIP config file
 
     Args:
         data (dict): raw YAML input from MIP analysis config file
 
     Returns:
         dict: parsed data
     """
+    mip_config = parse_obj_as(MipBaseConfig, data)
     return {
-        "email": data.get("email"),
-        "case": data.get("case_id") or data.get("family_id"),
-        "samples": get_sample_data_from_config(config=data),
-        "config_path": data.get("config_file_analysis"),
-        "is_dryrun": "dry_run_all" in data,
-        "log_path": data.get("log_file"),
-        "out_dir": data.get("outdata_dir"),
-        "priority": data.get("slurm_quality_of_service"),
-        "sampleinfo_path": data.get("sample_info_file"),
-    }
-
-
-def parse_sampleinfo_light(data: dict) -> dict:
-    """Parse MIP sample info file and retrieve mip_version, date, and status
-
-    Args:
-        data (dict): raw YAML input from MIP qc sample info file
-
-    Returns:
-        dict: {'version': str, 'date': str, 'is_finished': str}
-
-
-    """
-
-    outdata = {
-        "date": data.get("analysis_date"),
-        "version": data.get("mip_version"),
-        "is_finished": data.get("analysisrunstatus") == "finished",
+        "email": mip_config.email,
+        "case": mip_config.case_id or mip_config.family_id,
+        "samples": get_sample_data_from_config(config=mip_config),
+        "config_path": mip_config.config_file_analysis,
+        "is_dryrun": mip_config.dry_run_all,
+        "log_path": mip_config.log_file,
+        "out_dir": mip_config.outdata_dir,
+        "priority": mip_config.slurm_quality_of_service,
+        "sampleinfo_path": mip_config.sample_info_file,
     }
 
-    return outdata
 
-
-def get_rank_model_version(sample_info: dict, rank_model_type: str, step: str) -> str:
+def get_rank_model_version(sample_info: MipBaseSampleinfo, rank_model_type: str, step: str) -> str:
     """Get rank model version"""
-    for key in ("recipe", "program"):
-        if key in sample_info:
-            return sample_info[key][step][rank_model_type]["version"]
+    if sample_info.recipe:
```

Comment:
Try to make recipe and step into pydantic models. Remember they don't need to include everything from the original file, only the stuff that is being used within CG

Reply:
Done